### ISSUE TITLE
Travis upgrades for v1.0

### ISSUE THREAD

**Issue by The-Compiler** [qutebrowser/qutebrowser#2780]
With #2742, we'll also need various environment upgrades on Travis:

- [x] Drop `py34-cov` environment which uses Travis' Ubuntu Trusty (this also means we won't have to revert 069f908a61379373abd30f6975261337485659ec; segfaults recently started happening again there)
- [x] Introduce coverage in one of the other enviroments (probably with QtWebEngine and adjusted coverage pragmas)
- [ ] Update Debian Jessie to Debian Stretch
- [ ] Update Ubuntu Xenial to Ubuntu Zesty (17.04) or Artful (17.10) - though it's probably similar enough to Debian Stretch anyways (all Qt 5.7.1), so maybe just drop it.
- [ ] Maybe drop py35-pyqt5XX environments because we have the equivalent of py35-pyqt571 with Debian Stretch?
- [x] Check if we can use container environments (`sudo: false`) for all non-Docker ones
- [x] Switch to `lang: python`

**Comment by jgkamat** [comment]
I would also recommend you evaluate circleci.com, in my experience, it's been a LOT faster than travis, and has much better native docker support in circleci 2.0 (their new architecture). However, the one problem with circleci 2.0 is that it does not support forked PR's in their new configuration. If that's added, we should be able to get a similar setup to travis (have a matrix of builds, all running in parallel). It would even update to github with multiple status tokens (so you could see the multiple builds from github, not only from travis). 

If it's ever added, I'll let you know and maybe play around with making a circleci config file.

Tl;dr: circleci would be awesome and probably a lot faster, but right now it's crippled because the new (awesome) features that make it worthwile don't work on forked PR's.

**Comment by The-Compiler** [comment]
This is done now, except the old Debian/Ubuntu releases are still around. Right now I think it's easiest to just drop those and only keep Archlinux (for QtWebKit-NG) and the environments with PyQt from PyPI.
